src/lobject.h

```diff
@@ -54,19 +54,43 @@ typedef uint16_t LuaType;
 ** we care about from a performance point of view - if any
 ** other types appear then they are all treated as ANY
 **/
+
 typedef enum {
-  RAVI_TANY = 0,      /* Lua dynamic type */
-  RAVI_TNUMINT = 1,    /* integer number */
-  RAVI_TNUMFLT,        /* floating point number */
-  RAVI_TARRAYINT,      /* array of ints */
-  RAVI_TARRAYFLT,      /* array of doubles */
-  RAVI_TFUNCTION,      /* Lua or C Function */
-  RAVI_TTABLE,         /* Lua table */
-  RAVI_TSTRING,        /* string */
-  RAVI_TNIL,           /* NIL */
-  RAVI_TBOOLEAN,       /* boolean */
-  RAVI_TUSERDATA       /* userdata or lightuserdata */
-} ravitype_t;
+  RAVI_TI_NIL,
+  RAVI_TI_FALSE,
+  RAVI_TI_TRUE,
+  RAVI_TI_INTEGER,
+  RAVI_TI_FLOAT,
+  RAVI_TI_INTEGER_ARRAY,
+  RAVI_TI_FLOAT_ARRAY,
+  RAVI_TI_TABLE,
+  RAVI_TI_STRING,
+  RAVI_TI_FUNCTION,
+  RAVI_TI_USERDATA,
+  RAVI_TI_OTHER
+} ravi_type_index;
+
+typedef uint32_t ravi_type_map;
```

Comment:
Do we need this to be 32-bits?

Reply:
No, 16 bit would also work. However, the only place that this makes a difference is in `Upvaldesc`. And in all the other places it will be padded to at least 32 bit.